### DISCUSSION TITLE
FILTER OVERVIEW

### DISCUSSION THREAD

**Discussion by medhome63** [mse-org/mseuniverse#9] [Ideas]
Hi All,
My pupose is to propose another way of managing filter without criticizing the Martin's smart achievment. Also I wanted to use the very useful trick  of Fred ( published datalink ) to improve this. The only problem and not the least is to fire the tmsedatasource.ondatachanged even if the user input an esc or return key without changing field. Currently to filter a dataset we have do to :1. select filter 
2 type in the blanked form 
3 re-select 1 
4 find which field is selected  ( onendfilteredit I believe)
5 filter On

I think the steps 3 and 4 can be elimineted.

Thank you for participating

Med

[filter3.zip](https://github.com/user-attachments/files/18129619/filter3.zip)

**Comment by fredvs** [comment]
Hello Med.

I try your demo filter3.zip but when loading the project there is that error and if I skip it, the dbgrid is empty.
What is wrong?
![invalidedatastream](https://github.com/user-attachments/assets/fd0f56f4-4634-4f82-8a35-e939baf395c9)

**Comment by fredvs** [comment]
Re-hello Med.
OK, I did use **test.db** from /mseuniverse/samples/db/dbfilter/ and updated your zip, it is ok now:
[dbfilter3.zip](https://github.com/user-attachments/files/18140911/dbfilter3.zip)

But I am lost how to use mse-filter.
What are the steps, 1,2,3,4,5 that you explained in previous post, what icon must be used?
(I told you, I am a beginner in mse-db...)

![msedb](https://github.com/user-attachments/assets/7f5809dc-7120-4657-941b-ec9468cbb1f2)

**Comment by medhome63** [comment]
Hello ,
To try filter3 . Just right click tlabel on top to drop downmenu and select filter you want then push funnel button on dbnavigator after entering a value.

Thank for Help

Med

**Comment by medhome63** [comment]
[filter4.zip](https://github.com/user-attachments/files/18231454/filter4.zip)
NEW
